exp/addons/api/v1alpha3/clusterresourcesetbinding_types.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ANCHOR: ResourceBinding
+
+// ResourceBinding keeps status of a resource that is applied to the cluster that ClusterResourceSetBinding belongs to.
+type ResourceBinding struct {
+	// Hash is the hash of a resource's data. This can be used to decide if a resource is changed.
+	// For "ApplyOnce" ClusterResourceSet.spec.strategy, this is no-op as that strategy does not act on change.
+	Hash string `json:"hash,omitempty"`
```

Comment:
If this isn't used, might be worth removing it

Reply:
Even for the `ApplyOnce` strategy, `hash` can be indicative of if the same resource is applied to 2 different clusters or not. 

Also, we can add a `hash` field to the resources in ClusterResourceSet objects to see if a cluster has the latest version of the resource or not.

Reply:
How are we calculating the hash?

Reply:
ConfigMap/Secret may have multiple key-value pairs. Hash is computed by concatenating all individual sha256 hash of `values`.

Reply:
Can we store the hash on configmap & secrets? We could potentially have a different controller that's only responsible to hash these objects

Reply:
Yeah, instead of adding it to CRS resources, makes sense to add it as annotation.
But we still will need to keep the hash on CRSBinding objects.

Reply:
Added CRS as owner reference to resources.